### PR TITLE
[NT-2072] Backing From BackingFragment

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

Adapts a `Backing` from a `BackingFragment`.

**Note:** Added `TODO`s for work that will be done in subsequent PRs to keep them smaller. `Reward` is not handled yet in this PR.

# 🤔 Why

Continuation of the work to move our GraphQL requests to Apollo.

# 🛠 How

Added adaptation code for `Backing` from `BackingFragment` and tested it.

# ✅ Acceptance criteria

- [ ] A `Backing` can be created from a `BackingFragment`.

### CODE REVIEW

File: KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift [hunk 1]
```diff
@@ -0,0 +1,534 @@
+import Apollo
+@testable import KsApi
+import XCTest
+
+final class Backing_BackingFragmentTests: XCTestCase {
```
Comment (justinswart) [inline KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift:5]: For now we just achieve the same test coverage as we currently have in https://github.com/kickstarter/ios-oss/blob/main/KsApi/models/graphql/adapters/Backing+GraphBackingTests.swift

**Review by singhhari — APPROVED** [review]
lgtm

File: KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+    do {
+      var dict = backingDictionary()
+      dict["addOns"] = nil
+      dict["reward"] = nil
+
+      let fragment = try GraphAPI.BackingFragment(jsonObject: backingDictionary())
+      XCTAssertNotNil(fragment)
+
+      let backing = Backing.backing(from: fragment)
+      XCTAssertNotNil(backing)
```
Comment (msadoon) [inline KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift:28]: Maybe add a `TODO://` for `XCTAssertEqual(backing?.reward?.isNoReward, true)`

File: KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+    } catch {
+      XCTFail(error.localizedDescription)
+    }
+  }
+}
+
+private func backingDictionary() -> [String: Any] {
+  let json = """
+  {
```
Comment (msadoon) [inline KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift:37]: How can we generate these test strings in the future?

File: KsApi/models/graphql/adapters/User+UserFragment.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+extension User {
```
Comment (msadoon) [inline KsApi/models/graphql/adapters/User+UserFragment.swift:3]: Maybe add a `TODO:// Remove file once all instances of this function are replaced by User+UserFragment` in `User+GraphUser`

File: KsApi/models/graphql/adapters/Backing+BackingFragment.swift [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+extension Backing {
```
Comment (msadoon) [inline KsApi/models/graphql/adapters/Backing+BackingFragment.swift:3]: Maybe add a `TODO:// Remove file once all instances of this function are replaced by Backing+BackingFragment` in `Backing+GraphBacking`

**Review by msadoon — APPROVED** [review]
Good to go!

File: KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift (hunk 3 above)
Comment (justinswart) [reply KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift:37]: The way to do this is to run the query that Apollo produces in GraphiQL and copy the exact response to that query.

File: KsApi/models/graphql/adapters/User+UserFragment.swift (hunk 4 above)
Comment (justinswart) [reply KsApi/models/graphql/adapters/User+UserFragment.swift:3]: There will be a clean-up task to remove all old Graph* files once the Apollo integration is done.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| KsApi/models/graphql/adapters/Backing+BackingFragmentTests.swift:29:1 | warning: (spaceInsideComments) Add leading and/or trailing space inside comments. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**